test/cri-containerd/helpers/log.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"sync"
+
+	"github.com/Microsoft/go-winio"
+	"github.com/pkg/errors"
+)
+
+func main() {
+	var err error
+
+	defer func() {
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			os.Exit(1)
+		}
+	}()
```

Comment:
I would prefer that we make a helper function that returns err and we can handle it here. In general it's best to always use named return values if you plan on using a variable from the function in a defer statement and we can't do that here since we can't change the signature of main.

Reply:
like this?
```
def main() {
    if err := logger(); err != nil {
        fmt.Fprintln(os.Stderr, err)
        os.Exit(1)
    }
}

func logger() error {
    // the rest of the logic here
}
```